plutus-tx-plugin/test/Budget/9.2/gte0.budget.golden

```diff
@@ -0,0 +1,2 @@
+({cpu: 2210128053
+| mem: 6212402})
```

Comment:
As expected, defining this in terms of the new inlinable `all` makes this noticeably faster. Here are the stats with non-worker-wrapper-transformed `recursiveAll`:

```
({cpu: 2509519053
| mem: 7514102})
```

Reply:
Yeah, this makes a lot of difference and we should probably point it out in the doc.

Reply:
It's kinda "common sense". Pull arguments out of recursion if they never change during evaluation. Not only will you pass fewer arguments around, you'll also provide more opportunities for optimizing usages of those that don't change. Works in Plutus the same way it works in Haskell (except in Haskell you have to deal with rewrite rules, deforestation etc, which complicate the picture).

Reply:
We also have a ticket for doing this automatically: this is the "static argument transformation".